### PR TITLE
Feature/signing mail

### PR DESCRIPTION
**Pull request by meletios**

### CODE REVIEW

File: src/main/java/com/hedvig/productPricing/service/service/InsuranceNotificationServiceImpl.java [hunk 1]
```diff
@@ -16,30 +16,39 @@
...
-      String memberId, Boolean insuredAtOtherCompany, InsuranceCompaniesSE currentInsurer) {
-    if (currentInsurer.equals(InsuranceCompaniesSE.OTHER)) {
-      notifyUnknownInsurer(memberId);
+  public void sendInsuranceSignedNotification(
+    String memberId,
+    Boolean insuredAtOtherCompany,
+    InsuranceCompaniesSE currentInsurer) {
+
+    if (!insuredAtOtherCompany) {
+      notificationService.sendInsuranceSignedEmail(memberId, false, null);
```
Comment (johantjellden) [inline src/main/java/com/hedvig/productPricing/service/service/InsuranceNotificationServiceImpl.java]: We should tell notificaiton service that this has happend and let notification decide if we send an email. So rename it to something like **insurancesSigned**.

File: src/main/java/com/hedvig/productPricing/service/serviceIntegration/notificationService/NotificationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  private NotificationServiceClient notificationServiceClient;
+
+  public NotificationServiceImpl(NotificationServiceClient notificationServiceClient) {
+    this.notificationServiceClient = notificationServiceClient;
+  }
+
+  @Override
+  public void sendInsuranceSignedEmail(
+    String memberId,
+    boolean switchingFromCurrentInsurer,
```
Comment (johantjellden) [inline src/main/java/com/hedvig/productPricing/service/serviceIntegration/notificationService/NotificationServiceImpl.java]: Maybe this boolean is reduntant, we either have a currentInsurer or not.

File: src/main/java/com/hedvig/productPricing/service/sagas/ContractSignedSaga.java [hunk 3]
```diff
@@ -3,38 +3,53 @@
...
+    val productEntity = productRepository.findById(event.getProductId());
+
+    if (productEntity.isPresent()) {
+      ProductEntity product = productEntity.get();
+
+      InsuranceCompaniesSE insurer = InsuranceCompaniesSE.create(product.currentInsurer);
+      try {
+        insuranceNotificationService.sendInsuranceSignedNotification(
+          event.getMemberId(),
+          product.insuredAtOtherCompany && product.currentInsurer != null,
```
Comment (johantjellden) [inline src/main/java/com/hedvig/productPricing/service/sagas/ContractSignedSaga.java]: I think that we should treat **product.insuredAtOtherCompany == true ====> product.currentInsurer != null** as an invariant that cannot be broken. So maybe we should abort here and log an error, the point is that we don't want to send a welcome you insurance is active now to someone that has a existing insurance.

File: src/main/java/com/hedvig/productPricing/service/service/InsuranceNotificationServiceImpl.java (hunk 1 above)
Comment (meletios) [reply src/main/java/com/hedvig/productPricing/service/service/InsuranceNotificationServiceImpl.java]: :+1:

File: src/main/java/com/hedvig/productPricing/service/serviceIntegration/notificationService/NotificationServiceImpl.java (hunk 2 above)
Comment (meletios) [reply src/main/java/com/hedvig/productPricing/service/serviceIntegration/notificationService/NotificationServiceImpl.java]: i have change name, i agree it was too confusing

File: src/main/java/com/hedvig/productPricing/service/sagas/ContractSignedSaga.java (hunk 3 above)
Comment (meletios) [reply src/main/java/com/hedvig/productPricing/service/sagas/ContractSignedSaga.java]: added a check

**Pull request merged**